From 436c08639b7a4fc68060cbddf8ec9d06d2b3a54a Mon Sep 17 00:00:00 2001
From: Wincent Colaiuta <win@wincent.com>
Date: Sun, 15 Feb 2009 19:50:17 +0100
Subject: [PATCH] Fix autoloading of ActionController::Caching::Sweeper

Since dropping the explicit require of "action_controller/caching/sweeping",
evaluating a subclass of "ActionController::Caching::Sweeper" will
raise an "uninitialized constant" NameError.

So add another "autoload" declaration to ensure that this doesn't happen.

See also:

  http://groups.google.com/group/rubyonrails-talk/browse_thread/thread/323ff7ec2d95ee32
  http://rails.lighthouseapp.com/projects/8994/tickets/1977

Signed-off-by: Wincent Colaiuta <win@wincent.com>
---
 actionpack/lib/action_controller/caching.rb |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/actionpack/lib/action_controller/caching.rb b/actionpack/lib/action_controller/caching.rb
index 1d14df0..76fe20c 100644
--- a/actionpack/lib/action_controller/caching.rb
+++ b/actionpack/lib/action_controller/caching.rb
@@ -28,6 +28,7 @@ module ActionController #:nodoc:
     autoload :Fragments, 'action_controller/caching/fragments'
     autoload :Pages, 'action_controller/caching/pages'
     autoload :Sweeping, 'action_controller/caching/sweeping'
+    autoload :Sweeper, 'action_controller/caching/sweeping'
 
     def self.included(base) #:nodoc:
       base.class_eval do
-- 
1.6.1.3

